### PR TITLE
[PixArt Sigma] Support single file loading for PixArt Sigma

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

It's getting popular and future fine-tunes might benefit from this feature.

### CODE REVIEW

File: src/diffusers/loaders/single_file_model.py [hunk 1]
```diff
@@ -208,7 +214,7 @@ def from_single_file(cls, pretrained_model_link_or_path_or_dict: Optional[str] =
                     )
                 )
 
-            if isinstance(original_config, str):
+            if isinstance(original_config, str) and "PixArt" not in class_name:
```
Comment (sayakpaul) [inline src/diffusers/loaders/single_file_model.py:217]: PixArt doesn't have any official `yaml` like config like the SD ecosystem of the models do. This is the best I could think of while not introducing too many code changes. But I am open to other suggestions.

File: src/diffusers/loaders/single_file_utils.py [hunk 2]
```diff
@@ -1559,3 +1563,180 @@ def _legacy_load_safety_checker(local_files_only, torch_dtype):
...
+        checkpoint.pop("y_embedder.y_embedding")
+        checkpoint.pop("pos_embed")
+    except Exception as e:
+        logger.debug(f"Skipping {str(e)}")
+        pass
+
+    return converted_state_dict
+
+
+def create_diffusers_config_from_pixart(original_config, checkpoint, sample_size=None):
```
Comment (sayakpaul) [inline src/diffusers/loaders/single_file_utils.py:1716]: The `original_config` argument is unused in the function is here to support this call without failure:

https://github.com/huggingface/diffusers/blob/ec1aded12e8fbdc98d6703f9111ed824addebc77/src/diffusers/loaders/single_file_model.py#L216

File: tests/single_file/test_pixart_sigma_single_file.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            "generator": generator,
+            "num_inference_steps": 2,
+            "strength": 0.75,
+            "guidance_scale": 7.5,
+            "output_type": "np",
+        }
+        return inputs
+
+    def test_single_file_format_inference_is_same_as_pretrained(self, expected_max_diff=1e-4):
+        transformer = PixArtTransformer2DModel.from_single_file(self.ckpt_path, original_config=True)
```
Comment (sayakpaul) [inline tests/single_file/test_pixart_sigma_single_file.py:49]: Unlike other SD pipelines and models, PixArt models don't have any config which could be loaded. So, we provide a boolean here. But open to other suggestions.

**Comment by sayakpaul** [comment]
@DN6 a friendly ping here.

**Comment by DN6** [comment]
@sayakpaul Just for my reference could you point me to some Pixart Single File checkpoints that are widely used?

**Comment by sayakpaul** [comment]
Yeah when I added it I mistakenly thought a community uploaded checkpoint to be a single-file one. But it turned out to be `diffusers` and they didn't upload the config.json 🤷

Sorry for the trouble. I will close it but will keep the branch alive so that it can be revisited later if need be.

**Pull request closed without merging**